docs/book/src/topics/eks/ipv6-enabled-cluster.md

```diff
@@ -0,0 +1,81 @@
+
+# IPv6 Enabled Cluster
+
+CAPA supports IPv6 enabled clusters. Dual stack clusters are not yet supported, but
+dual VPC, meaning both ipv6 and ipv4 are defined, is support and in fact, it's the
+only mode of operation at the writing of this doc.
+
+Upcoming feature will be IPv6 _only_.
+
+## How to set up
+
+Two modes of operations are supported. Request AWS to generate and assign an address
+or BYOIP which is Bring Your Own IP. There must already be a provisioned pool and a
+set of IPv6 CIDR for that.
+
+### Automatically Generated IP
+
+To request AWS to assign a set of IPv6 addresses from an AWS defined address pool,
+use the following setting:
+
+```yaml
+kind: AWSManagedControlPlane
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+metadata:
+  name: "${CLUSTER_NAME}-control-plane"
+spec:
+  network:
+    vpc:
+      ipv6: {}
```

Comment:
Given you say the vpc has to be dualstack should there be some sort of ipv4 block here?

Reply:
You can define your ipv4 cidr as usual. They should work. This is dual stack in the sense that the vpc has both routes and not in the sense of kubernetes dual stack. :) So any setting that you add should work normally for IPv4.

Reply:
Yeah I think that makes sense, but I was thinking more in terms of if defining ipv4 is required or not.

Since the vpc needs to be dualstack I was thinking it makes sense to see:
```
vpc:
  ipv6: {}
  ipv4: {}
```

which can be differentiated from what I imagine an ipv6-only vpc would be (which I understand isn't going to be supported at the moment):
```
vpc:
   ipv6: {}
```

But this is me just wondering if it is important to make that distinction explicitly, perhaps it doesn't matter too much. If it doesn't happy to just consider this thread resolved.

Reply:
Yeah, I see what you're saying. However, IPv4 is set for you anyways internally. I think making it explicitly present in the VPC setting makes for poor readability. For ipv6 only this looks a lot better to me:

```yaml
vpc:
   ipv6:
   ipv6Only: true  
```

Which aligns more with the AWS settings anyways.

This is my opinion though. :)

Reply:
Cool, happy to defer to what you think makes sense in the end.

Reply:
It's a good question! :) I would like to ask the others what they think too. Some of it is behind the scenes. I'm not sure if it should be surfaced. But I defer to consensus too in the end. :)